tracer/src/Datadog.Trace/AppSec/Security.cs

```diff
@@ -246,10 +209,13 @@ private static void AddAppsecSpecificInstrumentations()
             {
                 Log.Debug("Sending CallTarget appsec derived integration definitions to native library.");
                 var payload = InstrumentationDefinitions.GetDerivedDefinitions(InstrumentationCategory.AppSec);
-                NativeMethods.InitializeProfiler(payload.DefinitionsId, payload.Definitions);
-                foreach (var def in payload.Definitions)
+                if (enable)
+                {
+                    NativeMethods.InitializeProfiler(payload.DefinitionsId, payload.Definitions);
+                }
+                else
                 {
-                    def.Dispose();
+                    NativeMethods.RemoveCallTargetDefinitions(payload.DefinitionsId, payload.Definitions);
                 }
 
                 Log.Information<int>("The profiler has been initialized with {count} AppSec derived definitions.", payload.Definitions.Length);
```

Comment:
this message should change to indicate if we are enabling or disabling

Reply:
Fixed in `73581a7a8289380ad851b8b4da85b66cbe880508`